### PR TITLE
feat: from_keystore method

### LINKED ISSUE #1368: dev: support keystore
Right now, afaik it's not possible to create a signer using a keystore path that would be great to improve security of backend services using starknet.py.

Happy to provide a PR if necessary.

## References
https://github.com/xJonathanLEI/starknet-rs/blob/016f3ee3293c8355f8a7cb64777adea6119493b9/starknet-signers/src/key_pair.rs#L60

### PR DESCRIPTION
**Pull request by EvolveArt**
Closes #1368 

## Introduced changes

- Adds https://pypi.org/project/eth-keyfile/ as a dependency
- Adds a `from_keystore` method to the `KeyPair` class

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: poetry.lock
Comment (ddoktorski) [inline poetry.lock:1]: Could you please update the dependencies using `poetry update eth-keyfile`? This way we can avoid updating dependencies that are not relevant for this change.

File: pyproject.toml
Comment (ddoktorski) [inline pyproject.toml:1]: What's the reason to format the `pyproject.toml` file?

File: starknet_py/net/signer/stark_curve_signer.py [hunk 1]
```diff
@@ -32,6 +34,18 @@ def from_private_key(key: Hash) -> "KeyPair":
             key = int(key, 0)
         return KeyPair(private_key=key, public_key=private_to_stark_key(key))
 
+    @staticmethod
+    def from_keystore(path: str, password: str) -> "KeyPair":
+        """
+        Create a key pair from a keystore file.
+        The keystore file should follow the Ethereum keystore format.
+        :param path: Path to the keystore file.
```
Comment (ddoktorski) [inline starknet_py/net/signer/stark_curve_signer.py:42]: A newline is necessary for correct formatting in the docs.
```suggestion
        The keystore file should follow the Ethereum keystore format.
        
        :param path: Path to the keystore file.
```

**Review by ddoktorski — CHANGES_REQUESTED** [review]
Thank you for your contribution! I've left a few minor comments that need to be addressed before merging :)

File: pyproject.toml
Comment (EvolveArt) [reply pyproject.toml:1]: just ran `poe format` in the repo

Comment (EvolveArt) [reply pyproject.toml:1]: reverted

File: poetry.lock
Comment (EvolveArt) [reply poetry.lock:1]: done

**Review by ddoktorski — APPROVED** [review]
Thank you! LGTM 🔥 

_Note: Tests on network fail due to unset secrets in the fork repository_

**Pull request merged**